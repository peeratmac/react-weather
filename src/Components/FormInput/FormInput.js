import React, { Component } from 'react';

class FormInput extends Component {
  handleGetWeather = event => {
    event.preventDefault();
    console.log('handleGetWeather Button Click!');
  };

  render() {
    return (
      <section className='form'>
        <h3>City or Country to Lookup. Go ahead!</h3>
        <form>
          <input
            name='city'
            type='text'
            value={''}
            placeholder='City'
            onChange={() => {}}
            className='form-input'
          />

          <input
            name='country'
            type='text'
            value={''}
            placeholder='Country'
            onChange={() => {}}
            className='form-input'
          />

          <button onClick={this.handleGetWeather} className='submit-button'>
            Get Weather
          </button>
        </form>
        <h3>Know the Lattitude and Longtitude? Try this.</h3>
        <input
          name='lattitude'
          type='text'
          value={''}
          placeholder='Lattitude'
          onChange={() => {}}
          className='form-input'
        />

        <input
          name='longtitude'
          type='text'
          value={''}
          placeholder='Longtitude'
          onChange={() => {}}
          className='form-input'
        />

        <button onClick={this.handleGetWeather} className='submit-button'>
          Get Weather
        </button>
        <form></form>
      </section>
    );
  }
}

export default FormInput;
