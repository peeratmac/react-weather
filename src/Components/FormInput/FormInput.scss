* {
  box-sizing: border-box;
  font-family: 'Rubik', sans-serif;
  text-align: center;
}

.form-section {
  background: url('https://www.metaweather.com/static/img/weather/sn.svg');
  background-repeat: no-repeat;
}
