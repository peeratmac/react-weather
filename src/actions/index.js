export const loadCities = cities => ({
  type: 'LOAD_CITIES',
  cities
});

export const targetCity = selectedCity => ({
  type: 'TARGET_CITY',
  selectedCity
});

export const getLatLong = latLong => ({
  type: 'GET_LAT_LONG',
  latLong
});

export const loadWeather = stationIDs => ({
  type: 'LOAD_WEATHER',
  stationIDs
});

export const setWeatherInfo = weatherInfo => ({
  type: 'SET_WEATHER_INFO',
  weatherInfo
});

export const getCurrentStation = currentStation => ({
  type: 'GET_CURRENT_STATION',
  currentStation
});

export const updateFavorties = favorites => ({
  type: 'UPDATE_FAVORITES',
  favorites
});
