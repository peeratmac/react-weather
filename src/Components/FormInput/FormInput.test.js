import React from 'react';
import { shallow } from 'enzyme';
import { FormInput } from './FormInput';
import { mapStateToProps, mapDispatchToProps } from './FormInput';
import { targetCity, getLatLong } from '../../actions';

describe('FORM INPUT', () => {
  let wrapper;
  const mockTargetCity = jest.fn();
  const mockGetLatLong = jest.fn();
  const mockSetWeatherInfo = jest.fn();
  const mockGetCurrentStation = jest.fn();
  const mockLoadCities = jest.fn();
  const mockLoadWeather = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <FormInput
        targetCity={mockTargetCity}
        getLatLong={mockGetLatLong}
        setWeatherInfo={mockSetWeatherInfo}
        getCurrentStation={mockGetCurrentStation}
        loadCities={mockLoadCities}
        loadWeather={mockLoadWeather}
        selectedCity={'Denver'}
        latLong={'1234,-2578'}
        stationIDs={[]}
        weatherInfo={[]}
        currentStation={47889}
        cities={[]}
      />
    );
  });

  it('should match the FormInput Snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it.skip('should call to send weatherInfo to redux store when the Send ALL Station Data button is clicked', () => {
    wrapper.instance().grabWeatherDataWithStationID();
    wrapper.instance().forceUpdate();
    expect(mockSetWeatherInfo).toHaveBeenCalled();
  });

  it.skip('should call to send more information with a click of button', () => {
    wrapper
      .find('button')
      .at(0)
      .simulate('click');

    expect(mockSetWeatherInfo).toHaveBeenCalled();
  });

  it('should show inital state of stationIDs available to be fetched', () => {
    const initialState = {
      stationIDs: [
        2487956,
        2487889,
        2488042,
        2487796,
        2488853,
        349859,
        56558361,
        773964,
        1132447,
        773692,
        2488867
      ]
    };

    expect(mapStateToProps(initialState).stationIDs).toEqual([
      2487956,
      2487889,
      2488042,
      2487796,
      2488853,
      349859,
      56558361,
      773964,
      1132447,
      773692,
      2488867
    ]);
  });

  it('should update selectedCity as the value changes in the form input', () => {
    const mockDispatch = jest.fn();

    const actionToDispatch = targetCity({
      selectedCity: 'Denver'
    });

    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.targetCity({
      selectedCity: 'Denver'
    });

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });

  it('should update latitude and longitude as the value changes in the form input', () => {
    const mockDispatch = jest.fn();

    const actionToDispatch = getLatLong({
      latLong: '54.977940,-1.611620'
    });

    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.getLatLong({
      latLong: '54.977940,-1.611620'
    });

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
