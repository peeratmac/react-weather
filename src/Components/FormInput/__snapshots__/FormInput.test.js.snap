// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FORM INPUT should match the FormInput Snapshot 1`] = `
<div
  className="form-section"
>
  <section
    className="form"
  >
    <h1>
      USE WEATHER STATION ID
    </h1>
    <form>
      <input
        className="form-input"
        name="station"
        onChange={[Function]}
        placeholder="Weather Station ID"
        type="text"
        value={47889}
      />
      <button
        className="submit-button"
        onClick={[Function]}
      >
        Send ALL Station Data
      </button>
      <button
        className="submit-button"
        onClick={[Function]}
      >
        Get Specific Weather
      </button>
    </form>
    <h1>
      CITY LOOKUP
    </h1>
    <form>
      <input
        className="form-input"
        name="city"
        onChange={[Function]}
        placeholder="City"
        type="text"
        value="Denver"
      />
      <button
        className="submit-button"
        onClick={[Function]}
      >
        Get Weather
      </button>
    </form>
    <h1>
      LATITUDE and LONGITUDE
    </h1>
    <form>
      <input
        className="form-input"
        name="lat-long"
        onChange={[Function]}
        placeholder="Latitude and Longitude"
        type="text"
        value="1234,-2578"
      />
      <button
        className="submit-button"
        onClick={[Function]}
      >
        Get Weather
      </button>
    </form>
  </section>
</div>
`;
